### DISCUSSION TITLE
Removing custom field data directly in the database

### DISCUSSION THREAD

**Discussion by markkuleinio** [netbox-community/netbox#6670] [Q&A] [answered]
I'm pretty sure that this has been discussed earlier somewhere but I could not find anything. Is it safe to modify `custom_field_data` fields directly in the database?

My case is that long time ago I had a custom field called "site_instance_id" assigned in the IP address objects but it was later unassigned from IP addresses. This meant the the existing data was still left in the objects. Now when I'm planning 2.8.5 -> 2.11.7 upgrade I noticed that editing the IP addresses in 2.11.7 is not possible if they have that old data in the database (sorry, didn't save the error message but it was maybe "unknown custom field 'site_instance_id'" or something like that).

So I went to the database and issued:

`update ipam_ipaddress set custom_field_data = custom_field_data::jsonb - 'site_instance_id' where custom_field_data::jsonb ? 'site_instance_id';`

and that removed the "site_instance_id" field and retained the other fields.

Now my question is: Is that safe to do, or should I do something else as well?

**Comment by candlerb** [comment, marked as answer]
I'm pretty sure that's all you need to do.

**Reply by markkuleinio** [reply]
Thanks!
